src/dune_rules/super_context.ml

```diff
@@ -541,6 +544,37 @@ let create ~(context : Context.t) ?host ~projects ~packages ~stanzas () =
       ~bin_artifacts_host:artifacts_host.bin ~find_package
   in
   let dune_dir_locations_var : Stdune.Env.Var.t = "DUNE_DIR_LOCATIONS" in
+  (* Add the site mentioned in stanza (could contain the external package) *)
+  let add_in_package_section m pkg section =
+    Package.Name.Map.update m pkg ~f:(function
+      | None -> Some (Section.Set.singleton section)
+      | Some s -> Some (Section.Set.add s section))
+  in
+  let package_sections =
+    List.fold_left stanzas ~init:Package.Name.Map.empty ~f:(fun acc stanzas ->
```

Comment:
I'm not certain, but perhaps this double fold can be replaced with `Dir_with_dune.deep_fold`.